functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/nodes/NodesSpec.groovy

```diff
@@ -0,0 +1,181 @@
+        nodesPage.partialTextIsPresent("password")
+    }
+
+    def "a nodes filter can be saved and used"() {
+        given:
+        final testFilterName = "test_filter_1"
+
+        nodesPage.setNodeInputText("name: password-node")
+        nodesPage.clickSearchNodes()
+        nodesPage.waitForNumberOfElementsToBeOne(nodesPage.nodeListTrBy)
+
+        when:
+        nodesPage.byAndWaitClickable(nodesPage.nodeFilterInputToggleBy).click()
+        nodesPage.byAndWaitClickable(nodesPage.nodeFilterInputDropdownSaveFilterBy).click()
+
+        nodesPage.setSaveNodeFilterModalNameFieldValue(testFilterName)
+        nodesPage.byAndWaitClickable(nodesPage.saveNodeFilterModalSaveButtonBy).click()
+
+        then: "filter is saved and used right away"
+        nodesPage.partialTextIsPresent(testFilterName)
+        nodesPage.waitForNumberOfElementsToBe(nodesPage.nodeListTrBy, 1)
+
+        when:
+        // Reset the filter to show all nodes
+        nodesPage.byAndWaitClickable(nodesPage.nodeFilterInputToggleBy).click()
+        nodesPage.byAndWaitClickable(By.partialLinkText("Show all nodes")).click()
+        nodesPage.waitForNumberOfElementsToBeMoreThan(nodesPage.nodeListTrBy, 1)
+
+        // Select the saved filter from the dropdown
+        nodesPage.byAndWaitClickable(nodesPage.nodeFilterInputToggleBy).click()
+        nodesPage.byAndWaitClickable(By.partialLinkText(testFilterName)).click()
+
+        then:
+        nodesPage.partialTextIsPresent(testFilterName)
+        nodesPage.waitForNumberOfElementsToBe(nodesPage.nodeListTrBy, 1)
+    }
+
+    def "a  command can be ran from the nodes filter"() {
+        given:
+        nodesPage.setNodeInputText("tags: \"ssh-node\"")
+        nodesPage.clickSearchNodes()
+        nodesPage.waitForNumberOfElementsToBe(nodesPage.nodeListTrBy, 2)
+
+        when:
+        nodesPage.byAndWaitClickable(nodesPage.actionsDropdownToggleBy).click()
+        nodesPage.byAndWaitClickable(nodesPage.actionsDropdownRunCommandBy).click()
+
+        then: "Ensure the filtered nodes appear on the page"
+        nodesPage.waitForUrlToContain("command/run")
+        nodesPage.partialTextIsPresent("2 Nodes Matched")
+        nodesPage.linkTextIsPresent("ssh-node")
+        nodesPage.linkTextIsPresent("password-node")
+    }
+
+    def "a  job can be created from the nodes filter"() {
+        given:
+        nodesPage.setNodeInputText("tags: \"ssh-node\"")
+        nodesPage.clickSearchNodes()
+        nodesPage.waitForNumberOfElementsToBe(nodesPage.nodeListTrBy, 2)
+
+        when:
+        nodesPage.byAndWaitClickable(nodesPage.actionsDropdownToggleBy).click()
+        nodesPage.byAndWaitClickable(nodesPage.actionsDropdownSaveJobBy).click()
+        JobCreatePage jobCreatePage = page JobCreatePage
+        jobCreatePage.byAndWaitClickable(By.partialLinkText("Nodes")).click()
+
+        then: "Ensure the filtered nodes appear on the page"
+        nodesPage.partialTextIsPresent("2 Nodes Matched")
```

Comment:
At this point are we sure `nodesPage` is referencing the Nodes tab on the job creation page instead of the original Node page reference?

Reply:
I've added the URL check to ensure we are in the right spot.